src/authentication/policies/OtpPolicy.tsx

```diff
@@ -0,0 +1,345 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Controller, useForm, useWatch } from "react-hook-form";
+import {
+  PageSection,
+  FormGroup,
+  Radio,
+  Select,
+  SelectVariant,
+  SelectOption,
+  NumberInput,
+  ActionGroup,
+  Button,
+  TextInput,
+  ButtonVariant,
+  AlertVariant,
+} from "@patternfly/react-core";
+
+import type RealmRepresentation from "@keycloak/keycloak-admin-client/lib/defs/realmRepresentation";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+import useToggle from "../../utils/useToggle";
+import { TimeSelector } from "../../components/time-selector/TimeSelector";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useRealm } from "../../context/realm-context/RealmContext";
+import { useAlerts } from "../../components/alert/Alerts";
+
+import "./otp-policy.css";
+
+const POLICY_TYPES = ["totp", "hotp"] as const;
+const OTP_HASH_ALGORITHMS = ["SHA1", "SHA256", "SHA512"] as const;
+const NUMBER_OF_DIGITS = [6, 8] as const;
+
+export const OtpPolicy = () => {
+  const { t } = useTranslation("authentication");
+  const {
+    control,
+    errors,
+    reset,
+    handleSubmit,
+    formState: { isDirty },
+  } = useForm({ mode: "onChange" });
+  const adminClient = useAdminClient();
+  const { realm: realmName } = useRealm();
+  const { addAlert, addError } = useAlerts();
+
+  const [open, toggle] = useToggle();
+  const [realm, setRealm] = useState<RealmRepresentation>();
+
+  const otpType = useWatch<typeof POLICY_TYPES[number]>({
+    name: "otpPolicyType",
+    control,
+    defaultValue: POLICY_TYPES[0],
+  });
+
+  useFetch(
+    () => adminClient.realms.findOne({ realm: realmName }),
+    (realm) => {
+      setRealm(realm);
+      reset({ ...realm });
+    },
+    []
+  );
+
+  const save = async (realm: RealmRepresentation) => {
+    try {
+      await adminClient.realms.update({ realm: realmName }, realm);
+      const updatedRealm = await adminClient.realms.findOne({
+        realm: realmName,
+      });
+      setRealm(updatedRealm);
+      reset({ ...updatedRealm });
+      addAlert(t("updateOtpSuccess"), AlertVariant.success);
+    } catch (error: any) {
```

Comment:
The type for `error` can be omitted here.
```suggestion
    } catch (error) {
```